### PR TITLE
Added first try of grammatics

### PR DESCRIPTION
**Pull request by jolex007**

### CODE REVIEW

File: docs/grammatic.txt [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+    <expression> |
+    <expression> "," <expression-list>
+
+<type> ::=
+    <type-identifier> |
+    "[" <type-identifier> "]"
+
+<type-identifier> ::=
+    <type-name> |
+    <type-name> "." <type-identifier>
```
Comment (Disadvantaged) [inline docs/grammatic.txt]: What does it mean? Is it for nested classes?

File: docs/grammatic.txt [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+
+<type-name> ::=
+    <identifier> |
```
Comment (Disadvantaged) [inline docs/grammatic.txt:97]: Probably, type-identifier?

File: docs/grammatic.txt [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+    "extension" <type-identifier> [ <type-inheritance-clause> ] <extension-body>
+
+<extension-body> ::=
+    "{" <extension-members> "}"
+
+<extension-members> ::=
+    <extension-member> [ <extension-members> ]
+
+<extension-member> ::=
+    <declaration>
```
Comment (Disadvantaged) [inline docs/grammatic.txt]: What should happen, when a new initializer is created inside extension?

File: docs/grammatic.txt [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+    ":" "[" <type> "]"
+
+<variable-declaration> ::=
+    "var" <pattern-initializer-list>
+
+<function-declaration> ::=
+    "func" <function-name> <function-declaration-argument-clause> [ <function-return-type> ] <code-block>
+
+<function-declaration-argument-clause> ::=
+    "(" [ <pattern-list> ] ")"
```
Comment (Disadvantaged) [inline docs/grammatic.txt:137]: Probably, function arguments should have been comma-separated?

File: docs/grammatic.txt [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+    "return" [ <expression> ]
+
+<expression> ::=
+    <primary-expression> |
+    <function-call-expression> |
+    <explicit-member-expression> |
+    <subcript-expression> |
+    <prefix-operator> <expression> |
+    <expression> <infix-operator> <expression> |
+    <array-expression>
```
Comment (Disadvantaged) [inline docs/grammatic.txt]: What does it mean?

File: docs/grammatic.txt [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+
+<array-expression> ::=
+    "[" <expression>* "]"
+
+<prefix-operator> ::=
+    "-" |
+    "!"
```
Comment (Disadvantaged) [inline docs/grammatic.txt]: Shouldn't we have "not" operator if we already have "and" and "or" operators?

File: docs/grammatic.txt [hunk 7]
```diff
@@ -0,0 +1,225 @@
...
+
+<infix-operator> ::=
+    "+" | "*" | "/" | "+" | "and" | "or" | "="
```
Comment (Disadvantaged) [inline docs/grammatic.txt]: Type with 2 "+" ops;
Is "=" a comparison operator or assignment operator? 
We should probably have both, or even better leave assignment as a statement

Comment (Disadvantaged) [reply docs/grammatic.txt]: Also there are no other comparison operators

**Review by Disadvantaged — CHANGES_REQUESTED** [review]

File: docs/grammatic.txt (hunk 2 above)
Comment (jolex007) [reply docs/grammatic.txt:97]: I've deleted this nterm

File: docs/grammatic.txt (hunk 1 above)
Comment (jolex007) [reply docs/grammatic.txt]: Yes, it is for nested classes or types
For example:
```
class A {
 class B;
}

A.B
```

File: docs/grammatic.txt (hunk 4 above)
Comment (jolex007) [reply docs/grammatic.txt:137]: Fixed

File: docs/grammatic.txt (hunk 5 above)
Comment (jolex007) [reply docs/grammatic.txt]: It is used to address for array like ```a[5]```

File: docs/grammatic.txt (hunk 6 above)
Comment (jolex007) [reply docs/grammatic.txt]: Fixed

File: docs/grammatic.txt (hunk 7 above)
Comment (jolex007) [reply docs/grammatic.txt]: = is for assignment
It is an expression, because we want to write like ```a = b = c```

Comment (jolex007) [reply docs/grammatic.txt]: Other comparison operators added

File: docs/grammatic.txt [hunk 8]
```diff
@@ -0,0 +1,236 @@
...
+
+<array-expression> ::=
+    "[" <expressions> "]"
+
+<prefix-operator> ::=
+    "-" |
+    "not"
+
+<infix-operator> ::=
+    "+" | "*" | "/" | "-" | "and" | "or" | "=" | ">" | ">=" | "<" | "<=" | "=="
```
Comment (Disadvantaged) [inline docs/grammatic.txt:211]: There is "!=" sign missing.

**Review by Disadvantaged — APPROVED** [review]

**Pull request merged**